From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: David Llewellyn-Jones <david@flypig.co.uk>
Date: Wed, 9 Aug 2023 23:30:20 +0100
Subject: [PATCH 03/67] Fix embedlite building. JB#50505

Updates the build configuration for embedlite.

Adds the --with-embedlite flag so that we can use the flag in
embedding/embedlite/config/mozconfig.merqtxulrunner.

Adds the --enable-dconf flag back to allow the build to continue.

Co-authored-by: Raine Makelainen <raine.makelainen@jolla.com>
---
 toolkit/moz.configure | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/toolkit/moz.configure b/toolkit/moz.configure
index 3b06076c38df8..9bb5a7251a471 100644
--- a/toolkit/moz.configure
+++ b/toolkit/moz.configure
@@ -3489,6 +3489,14 @@ option(env="MOZ_CRASHREPORTER_MOCK", help="Mock the crashreporter to test native
 set_config("MOZ_CRASHREPORTER_MOCK", True, when="MOZ_CRASHREPORTER_MOCK")
 
 
+# Embedlite
+# ==============================================================
+option("--with-embedlite", help="Build with embedlite")
+
+set_config("MOZ_EMBEDLITE", True, when="--with-embedlite")
+set_define("MOZ_EMBEDLITE", True, when="--with-embedlite")
+
+
 # Wine
 # ==============================================================
 @depends(target, host)
@@ -3608,6 +3616,22 @@ with only_when(toolkit_gtk | toolkit_qt):
         set_define("MOZ_ENABLE_DBUS", True)
 
 
+# dconf
+# ==============================================================
+with only_when(toolkit_gtk | toolkit_qt):
+    option("--disable-dconf", help="Disable dconf support", when=use_pkg_config)
+
+    @depends("--enable-dconf", when=use_pkg_config)
+    def enable_dconf_option(enable_dconf):
+        return enable_dconf
+
+    with only_when(enable_dconf_option):
+        pkg_check_modules("MOZ_DCONF", "dconf")
+
+        set_config("MOZ_ENABLE_DCONF", True)
+        set_define("MOZ_ENABLE_DCONF", True)
+
+
 # Necko's wifi scanner
 # ==============================================================
 @depends(target)
